File: 1-easy/problems/1331-rank-transform-of-an-array/solution.ts
function arrayRankTransform(arr: any): boolean | number | string | any {
  """
  :type arr: List[int]
  :rtype: List[int]
  """
  return map({x: i+1 for i, x in enumerate(sorted(set(arr)))}.get, arr);
}

export { arrayRankTransform };

File: 1-easy/problems/1331-rank-transform-of-an-array/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def arrayRankTransform(self, arr):
        """
        :type arr: List[int]
        :rtype: List[int]
        """
        return map({x: i+1 for i, x in enumerate(sorted(set(arr)))}.get, arr)
